datadog_lambda/wrapper.py

```diff
@@ -382,9 +383,31 @@ def _after(self, event, context):
             logger.error(format_err_with_traceback(e))
 
 
+class _ErrorFallbackHandler(object):
+    """
+    Fallback handler for when an exception occurs outside of the handler function.
+    Emits telemetry and re-raises the exception.
+    """
+
+    def __init__(self, exception, modified_mod_name, start_time_ns):
+        self.exception = exception
+        self.modified_mod_name = modified_mod_name
+        self.start_time_ns = start_time_ns
+
+    def __call__(self, event, context, **kwargs):
+        emit_telemetry_on_exception_outside_of_handler(
+            context,
+            self.exception,
+            self.modified_mod_name,
+            self.start_time_ns,
+        )
+        raise self.exception
+
+
 def format_err_with_traceback(e):
     tb = traceback.format_exc().replace("\n", "\r")
     return f"Error {e}. Traceback: {tb}"
 
 
 datadog_lambda_wrapper = _LambdaDecorator
+error_fallback_handler = _ErrorFallbackHandler
```

Comment:
Not to belabor the point too much, but wrapper/decorator/handler, all words we use to refer to the same thing? That's potentially confusing for our future selves. Honestly, I would prefer we match the existing naming as much as we can, however wrong it may be.

Therefore, this line would become
```py
error_fallback_wrapper = _ErrorFallbackDecorator
```

But, also this is really nitpicky, so it's fine to keep it as is too.

Reply:
This is not a wrapper/decorator, that's why I suggested to not name it as a Decorator!

Reply:
Ah yes, I see what you mean. We are not passing the original handler to this `_ErrorFallbackHandler` and thus it is not a wrapper/decorator. Semantics. No strong opinion from me on what we should do here, I'm fine if we keep it as is.